src/base/snark0.ml

```diff
@@ -2138,30 +2148,24 @@ module Run = struct
       a
 
     let inject_wrapper :
-        type r_var r_value k_var k_value.
-           (r_var, r_value, k_var, k_value) Data_spec.t
-        -> f:(r_var -> r_var)
-        -> k_var
-        -> k_var =
-     fun spec ~f ->
-      match spec with
-      | Data_spec _ ->
-          fun x a ->
-            let inject_wrapper ~f x = f x in
-            inject_wrapper ~f (x a)
-
-    let constraint_system ~exposing ~return_typ x =
-      let x = inject_wrapper exposing x ~f:(fun x () -> mark_active ~f:x) in
-      Perform.constraint_system ~run:as_stateful ~exposing ~return_typ x
+        type r_var input_var.
+        f:(r_var -> r_var) -> (input_var -> r_var) -> input_var -> r_var =
+     fun ~f x a ->
+      let inject_wrapper ~f x = f x in
+      inject_wrapper ~f (x a)
```

Comment:
```ocaml
let inject_wrapper ~f circuit:(input_var -> return_var) input_var : return_var =
  f (circuit input_var)
in
inject_wrapper
```

would this work? perhaps clearer

Reply:
Probably is, but leaving this PR as the minimum possible to remove `Data_spec.t`. Feel free to open another PR for cleanups etc.